pkg/kubernetes/details.js

```diff
@@ -3,12 +3,130 @@ define([
+        var ns = $(e.relatedTarget).attr("data-ns");
+        var entity = k8client.lookup(kind, name, ns);
+        delete_entity_dlg.find('.modal-body').text("Delete " + kind + " " + name + "?");
+
+        delete_btn.on('click', function() {
+            k8client.remove(entity.metadata.selfLink);
+            delete_entity_dlg.modal('hide');
+        });
+    });
+
+    adjust_entity_dlg.on('show.bs.modal', function(e) {
+        var name = $(e.relatedTarget).attr("data-name");
+        var kind = $(e.relatedTarget).attr("data-kind");
+        var ns = $(e.relatedTarget).attr("data-ns");
+        var entity = k8client.lookup(kind, name, ns);
+
+        adjust_entity_dlg.find('#replica-count').val(entity.spec.replicas);
+
+        adjust_btn.on('click', function() {
+            function resize(item, value) {
+                var spec = item.spec;
+                if (!spec) {
+                    console.warn("replicationcontroller without spec");
+                    return false;
+                }
+
+                /* Already set at same value */
+                if (spec.replicas === value)
+                    return false;
+
+                spec.replicas = value;
+                return true;
+            }
+
+            function update_replica(input, rc) {
+                var failures = [];
+                var dfd = $.Deferred();
+                var req;
+                var ex;
+
+                var value = Number($.trim(input));
+                if (isNaN(value) || value < 0)
+                    ex = new Error(_("Not a valid number of replicas"));
+                else if (value > 128)
+                    ex = new Error(_("The maximum number of replicas is 128"));
+
+                if (ex) {
+                    ex.target = "#replica-count";
+                    failures.push(ex);
+                }
+
+                var task = function() {
+                    return k8client.modify(rc.metadata.selfLink, function(item) {
+                        return resize(item, value);
+                    });
+                };
+
+                task.label = cockpit.format(_("Updating $0..."), rc.metadata.name);
+
+                if (failures.length) {
+                    adjust_entity_dlg.dialog("failure", failures);
+                    dfd.reject(ex);
+                }
+
+                dfd.notify(task.label || null);
+
+                req = task()
+                    .done(function() {
+                        dfd.resolve();
+                    })
+                    .fail(function(ex) {
+                        dfd.reject(ex);
+                    });
+                var promise = dfd.promise();
+                promise.cancel = function cancel() {
+                    if (req && req.cancel)
+                        req.cancel();
+                };
+
+                return promise;
+            }
+
+            var new_value = adjust_entity_dlg.find('#replica-count').val();
```

Comment:
Why are we abstracting this into a separate function? There doesn't seem to be a need for this additional layer.